api/src/main/java/edu/cornell/mannlib/vitro/webapp/web/directives/DataGetterDirective.java

```diff
@@ -0,0 +1,139 @@
+/* $This file is distributed under the terms of the license in LICENSE$ */
+
+package edu.cornell.mannlib.vitro.webapp.web.directives;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import edu.cornell.mannlib.vitro.webapp.controller.VitroRequest;
+import edu.cornell.mannlib.vitro.webapp.utils.dataGetter.DataGetter;
+import edu.cornell.mannlib.vitro.webapp.utils.dataGetter.DataGetterUtils;
+import freemarker.core.Environment;
+import freemarker.template.ObjectWrapper;
+import freemarker.template.TemplateDirectiveBody;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateModel;
+import freemarker.template.TemplateModelException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.jena.ontology.OntModel;
+
+/**
+ * Freemarker directive to make substitutions in DataGetter and return the data
+ * in variable.
+ */
+public class DataGetterDirective extends BaseTemplateDirectiveModel {
+    private static final Log log = LogFactory.getLog(DataGetterDirective.class);
+
+    @Override
+    public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body)
+            throws TemplateException, IOException {
+        long startTime = System.nanoTime();
+        String dataGetterUri = getRequiredSimpleScalarParameter(params, "uri");
+        String variableName = getOptionalSimpleScalarParameter(params, "var");
+        HttpServletRequest req = (HttpServletRequest) env.getCustomAttribute("request");
+        VitroRequest vreq = new VitroRequest(req);
+        debug(getTimeSince(startTime) + "ms .1");
+        try {
+            debug(getTimeSince(startTime) + "ms .2");
+            OntModel model = vreq.getDisplayModel();
+            debug(getTimeSince(startTime) + "ms .3");
+            DataGetter dataGetter = DataGetterUtils.dataGetterForURI(vreq, model, dataGetterUri);
+            debug(getTimeSince(startTime) + "ms .4");
+            Map<String, Object> parameters = getOptionalHashModelParameter(params, "parameters");
+            debug(getTimeSince(startTime) + "ms .5");
+            applyDataGetter(dataGetter, env, parameters, variableName);
+            debug(getTimeSince(startTime) + "ms .6");
```

Comment:
It is great that the implementation is quite good in performance. Although it is logged only in debug mode, I think we should remove those time consumption logging. What do you think? We might keep one at the end if we want to leave options for VIVO customers to check efficiency of some Sparql query behind data getter, but I am not sure it is necessary.

Reply:
I removed some not needed log messages. Improved messages for those that could be useful for debugging purposes.